### PR TITLE
deps: Vendor github.com/aws/aws-sdk-go/service/route53domains

### PR DESCRIPTION
**Pull request by gazoakley**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #88

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ go test ./aws
ok  	github.com/terraform-providers/terraform-provider-aws/aws	15.003s
```

### CODE REVIEW

**Comment by gazoakley** [comment]
This will allow me to update #12711

File: .hashibot.hcl [hunk 1]
```diff
@@ -425,7 +425,7 @@ behavior "regexp_issue_labeler_v2" "service_labels" {
       "aws_route53_([^d]|d[^o]|do[^m]|dom[^a]|doma[^i]|domai[^n]|domain[^s]|domains[^_]|[^r]|r[^e]|re[^s]|res[^o]|reso[^l]|resol[^v]|resolv[^e]|resolve[^r]|resolver[^_])",
     ],
     "service/route53domains" = [
-      "aws_route53_domains_",
+      "aws_route53_domain",
```
Comment (bflad) [inline .hashibot.hcl:428]: Our preference for naming these resources will be `aws_route53domains_THING`, e.g. `aws_route53domains_domain`

Comment (gazoakley) [reply .hashibot.hcl:428]: For consistency with Route53 Resolver - wouldn't it then be better to go with `aws_route53_domains_domain`? (although that feels a bit repetitive too 😄)

Comment (bflad) [reply .hashibot.hcl:428]: Going forward, we'd really like to get in the habit of `aws_SERVICE_THING` to more align with CloudFormation resources as well as prevent any oddities of V2 APIs. See also: https://github.com/terraform-providers/terraform-provider-aws/pull/8842#discussion_r390546009

Comment (gazoakley) [reply .hashibot.hcl]: Updated to `aws_route53domains_` to match service name

**Pull request merged**